### PR TITLE
DL-396 python package codeartifact

### PR DESCRIPTION
**Pull request by tom-linius**
- On merge to main, build and publish this package to the CodeArtifact repository in the `prod` account
- pytest run on non main branches
- github workflows removed as CircleCI is CI tool of choice.

### CODE REVIEW

File: pypi_auth.sh [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    --domain-owner "$DOMAIN_OWNER" \
+    --domain "$DOMAIN" \
+    --profile "$PROFILE" \
+    --region "$REGION" \
+    --query authorizationToken \
+    --output text
+)
+
+export PIP_EXTRA_INDEX_URL="https://aws:$AUTH_TOKEN@$DOMAIN-$DOMAIN_OWNER.d.codeartifact.$REGION.amazonaws.com/pypi/$REPO/simple"
+echo "$PIP_EXTRA_INDEX_URL"
```
Comment (CygnusAlpha) [inline pypi_auth.sh:23]: newline. The export won't work unless you're `source`ing the file btw.

Comment (CodePint) [reply pypi_auth.sh:23]: I think the export is for if you want to source the file and login/publish manually (we have done this in other packages)

Comment (tom-linius) [reply pypi_auth.sh:23]: Yep just been copying and pasting this helper script around

Comment (CodePint) [reply pypi_auth.sh:23]: lets just leave it as it is useful

**Review by CodePint — APPROVED** [review]

File: setup.py [hunk 2]
```diff
@@ -1,57 +1,61 @@
...
+    include_package_data=True,
+    install_requires=load_requirements(),
+    python_requires="~=3.8",
     extras_require={
-        "dev": dev_requirements,
-        "ci": ["pytest", "pytest-cov"],
-        "doc": doc_requirements,
+        "dev": [
+            "tox==3.5.3",
+            "black==20.8b1",
```
Comment (CodePint) [inline setup.py]: black requirement should be removed (just noticed)

**Comment by AlanDonohoe** [comment]
Could someone give a little more context please?

We are hosting our own forked celery-director as a package in AWS' CodeArtifact?

And then we can install this in work flow engine?

**Review by CygnusAlpha — APPROVED** [review]

**Comment by tom-linius** [comment]
@AlanDonohoe Yep you're bang on - forked celery-director in our own pypi repo in AWS CodeArtifact. 

So far we have been referencing this repository directly in the requirements.txt of WFE (https://github.com/LiniusAustPty/workflow-engine/blob/fa37b7c0b9da8a6c45bf279034ecd6a30ecbcdcc/requirements.txt#L8) and once merged, we'll be able to replace that with `celery-director-linius` (George just added the -linius, makes sense)

We've done some work in pylvs and sqla_oauth to automate the publishing to CodeArtifact, so this is just a continuation.

**Comment by CodePint** [comment]
> Could someone give a little more context please?
> 
> We are hosting our own forked celery-director as a package in AWS' CodeArtifact?
> 
> And then we can install this in work flow engine?

Exactly. Nail on the head :)

**Pull request merged**